jsx/brisk_ppx.ml

```diff
@@ -76,24 +76,19 @@ module JSX_ppx = struct
 end
 
 module Declaration_ppx = struct
-  let func_pattern = Ppxlib.Ast_pattern.(pexp_fun __ __ __ __)
-
-  let match_ pattern ?on_error loc ast_node ~with_ =
-    Ppxlib.Ast_pattern.parse pattern ?on_error loc ast_node with_
-
   let attribute_name = function
     | `Component -> "component"
     | `Native -> "nativeComponent"
 
   let transform_component_expr ~useDynamicKey ~attribute ~component_name expr =
     let rec map_component_expression ({ P.pexp_loc = loc } as expr) =
-      match_ func_pattern loc expr
-        ~with_:(fun lbl opt_arg pat child_expression ->
+      match expr with
+      | {pexp_desc = Pexp_fun (lbl, opt_arg, pat, child_expression)} ->
```

Comment:
I wanted to avoid pattern matching on the AST because it's too fragile. Ppxlib uses OMP behind the scenes

Reply:
Right. I wish the new ppx tools were ready so I didn't have to learn ppxlib and use its broken documentation. I'll see if I can figure out how to do more complex pattern matching with it though.

Reply:
I'll help u

Reply:
https://github.com/dune-universe/dune-universe/blob/9aedbe8d7566abb87897ae7e6c564466ef8bd79c/packages/ocamlapi.0.0.2/ocamlapi_ppx/src/route_ast_helpers.ml#L8

@glennsl there you go.

Reply:
I also had to map the patterns to use a common variant type to distinguish them and since they don't have the same argument types. But it was actually pretty straightforward.